src/clients/node/src/node.zig

```diff
@@ -579,7 +579,11 @@ fn request(env: c.napi_env, info: c.napi_callback_info) callconv(.C) c.napi_valu
     const context = contextCast(context_raw.?) catch return null;
     const operation_int = translate.u32_from_value(env, argv[1], "operation") catch return null;
 
-    if (operation_int >= @typeInfo(Operation).Enum.fields.len) {
+    const greatest_enum_value = @enumToInt(@field(
+        Operation,
+        @typeInfo(Operation).Enum.fields[@typeInfo(Operation).Enum.fields.len - 1].name,
+    ));
+    if (operation_int > greatest_enum_value) {
```

Comment:
I think you could use `@intToEnum(vsr.Operation, operation).valid(StateMachine)`, which is a little more precise of a check.

Reply:
Interesting, I wasn't sure what would happen if you tried casting an integer outside of the enum value space. If it would crash or something.

Reply:
I _think_ in this case it won't only because `vsr.Operation` is a non-exhaustive enum (note the `_,` variant): 

https://github.com/tigerbeetledb/tigerbeetle/blob/e5bf5393e7a658cf5ca6fc4d7291cedc83bf9ec6/src/vsr.zig#L126-L137

And https://ziglang.org/documentation/master/#toc-Non-exhaustive-enum:

> [@intToEnum](https://ziglang.org/documentation/master/#intToEnum) on a non-exhaustive enum involves the safety semantics of [@intCast](https://ziglang.org/documentation/master/#intCast) to the integer tag type, but beyond that always results in a well-defined enum value.

(But I haven't tested it)

Reply:
Cool, got it working and also tested that it still catches an invalid operation. Thanks!